pkg/cloudprovider/cloudapi/azure/azure_nsg_rules.go

```diff
@@ -581,9 +637,11 @@ func convertFromAzureProtocolToNepheControllerProtocol(azureProtoName *armnetwor
 		return nil, nil
 	}
 
-	protocolNum, found := azureProtoNameToNumMap[*azureProtoName]
+	// direct modification on Azure portal will cause the protocol name change, not matching Azure own constants, so lowercase is needed.
+	// e.g. Tcp will change to TCP after portal modify, even if protocol is not touched at all.
```

Comment:
what do you mean by this.. If NSG is modified out of band, it will update Nephe rules? If so, drop portal, because rules can be updated via CLI or console. So just mention 'out of band'.

Reply:
This is an Azure issue. So basically if user modify the rule on portal, the protocol name will change to all caps even if user didn't modify that field. For example, the rule Nephe creates will have property `"protocol": "Tcp"`, which is from Azure's const. However if user modify the rule on the portal, even if user only modifies the description, that protocol will change to `"protocol": "TCP"`, causing parsing issue since we were using Azure const for matching. All rule created on portal will also be all caps in protocol. This is the reason I'm converting it to all lower case on matching.

Reply:
ok.. change it to out-of-band. rule update  can happen from multiple places

Reply:
Done.